src/cli/self_update/unix.rs

```diff
@@ -57,66 +57,50 @@ pub fn complete_windows_uninstall() -> Result<utils::ExitCode> {
     panic!("stop doing that")
 }
 
-pub fn do_remove_from_path(methods: &[PathUpdateMethod]) -> Result<()> {
-    for method in methods {
-        if let PathUpdateMethod::RcFile(ref rcpath) = *method {
-            let file = utils::read_file("rcfile", rcpath)?;
-            let addition = format!("\n{}\n", shell_export_string()?);
+pub fn do_remove_from_path() -> Result<()> {
+    for sh in shell::get_available_shells() {
+        let source_bytes = format!("\n{}\n", sh.source_string()?).into_bytes();
 
+        // Check more files for cleanup than normally are updated.
+        for rc in sh.rcfiles().iter().filter(|rc| rc.is_file()) {
+            let file = utils::read_file("rcfile", &rc)?;
             let file_bytes = file.into_bytes();
-            let addition_bytes = addition.into_bytes();
-
-            let idx = file_bytes
-                .windows(addition_bytes.len())
-                .position(|w| w == &*addition_bytes);
-            if let Some(i) = idx {
-                let mut new_file_bytes = file_bytes[..i].to_vec();
-                new_file_bytes.extend(&file_bytes[i + addition_bytes.len()..]);
-                let new_file = String::from_utf8(new_file_bytes).unwrap();
```

Comment:
one additional thing I missed: this change will no longer uninstall previous installs of rustup properly; I think checking for the prior form of rustup path add would make sense.

Reply:
I agree. Kinnison said I didn't _have_ to do that, but I was going to anyways, and mostly wanted to see an initial review pass on this PR before doing more work (in case I had to throw everything out or something).

It's going to be a distinct function so that it doesn't get tangled in the shell-handling logic, and it will be invoked both coming and going.